contracts/base/OnlySelfCallable.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+
+abstract contract OnlySelfCallable {
+    error InvalidSender(address sender);
+    error InvalidCalldataLength();
+
+    modifier onlySelf() {
+        checkOnlySelf();
+        _;
+    }
+
+    function checkOnlySelf() private view {
```

Comment:
I find this name very misleading, as `address(this)` has no impact on this check.

Reply:
then `checkCallerisSender()`?

Reply:
updated to `onlyAccount`